setup.py

```diff
@@ -27,8 +27,18 @@
 # prerequisite: setuptools
 # http://pypi.python.org/pypi/setuptools
 
+EXTRAS = {}
+REQUIRES = []
 with open('requirements.txt') as f:
-    REQUIRES = f.readlines()
+    for line in f:
+        line, _, _ = line.partition('#')
+        line = line.strip()
+        if ';' in line:
+            requirement, _, specifier = line.partition(';')
+            for_specifier = EXTRAS.setdefault(':{}'.format(specifier), [])
+            for_specifier.append(requirement)
```

Comment:
what is the purpose of "for_specifier.append(requirement)" but for_specifier is not kept anywhere?

Reply:
`for_specifier` is the list inside `EXTRAS[f":{spec}"]` -- I'd use `collections.defaultdict(list)` here but setuptools requires specific types here.

Reply:
The trick here is [`setdefault`](https://docs.python.org/3/library/stdtypes.html#dict.setdefault)